### PR TITLE
Remove legacy building.Locked and building.BuildComplete from Gate.

### PR DESCRIPTION
**Pull request by pchote**
Another step towards eliminating these legacy code warts.

### CODE REVIEW

**Comment by pchote** [comment]
Re (lack of) update rules: we have a few traits left to do, with a lot of that work already sitting open in parallel PRs.  Most of the rule changes also span multiple levels of inheritance, so i'm not sure we can even usefully show a list of traits to change.  If we don't have any better ideas I think our best option is to make a general note about this on https://github.com/OpenRA/OpenRAModSDK/wiki/Update-notes:-release-20180923-to-next

File: OpenRA.Mods.Common/Traits/Buildings/Gate.cs [hunk 1]
```diff
@@ -34,7 +34,7 @@ public class GateInfo : PausableConditionalTraitInfo, ITemporaryBlockerInfo, IBl
 	}
 
 	public class Gate : PausableConditionalTrait<GateInfo>, ITick, ITemporaryBlocker, IBlocksProjectiles,
-		INotifyAddedToWorld, INotifyRemovedFromWorld, INotifyBlockingMove, ISync
```
Comment (MustaphaTR) [inline OpenRA.Mods.Common/Traits/Buildings/Gate.cs]: There is a value with `[Sync]` (Position), why remove `ISync`? Tbh this doesn't look related so i guess this was a mistake.

Comment (MustaphaTR) [reply OpenRA.Mods.Common/Traits/Buildings/Gate.cs]: Makes sense.

**Review by MustaphaTR — DISMISSED** [review]
Didn't notice any regressions about Gates in TS mod.

**Comment by pchote** [comment]
Renamed `makeanimation-active` to `build-incomplete` to match #15682.

**Review by abcdefg30 — APPROVED** [review]
